scripts/cli/src/util/checkBridgeBalance.ts

```diff
@@ -0,0 +1,164 @@
+import { ApiSingleton } from "../util/init";
+import PrettyError from "pretty-error";
+import * as prom from "./promClient.js";
+import express from "express";
+
+/// Express
+const app = express();
+const port = 8888;
+
+async function main(): Promise<void> {
+	type userInfo = {
+		polyx: number;
+		lastTimeBridged: number;
+	};
+	prom.injectMetricsRoute(app);
+	prom.startCollection();
+	console.log(
+		`Starting events monitor watching endpoint: ${process.env.WS_PROVIDER}`
+	);
+	app.listen(port, () =>
+		console.log(`Offences monitor running on port ${port}`)
+	);
+	const api = await ApiSingleton.getInstance();
+	const BRIDGE_LIMIT = 500_000_000;
+	const BRIDGE_POLY_TOTAL = 1_000_000_000;
+	const BRIDGE_TIME_PERIOD = await blocksToSeconds();
+	let userArray: Map<string, userInfo> = new Map();
+
+	async function blocksToSeconds() {
+		const timelock = await api.query.bridge.timelock();
+		return timelock.toNumber() * 6;
+	}
+
+	function currentTimeToSeconds() {
+		const currentDateTime = new Date();
+		const resultInSeconds = currentDateTime.getTime() / 1000;
+		return resultInSeconds;
+	}
+
+	function hoursToSeconds(hours: number) {
+		let seconds = hours * 3600;
+		return seconds;
+	}
+
+	function checksUser(
+		userMap: Map<string, userInfo>,
+		newPolyx: any,
+		offenderAddress: any,
+		blockNumber: number
+	) {
+		const user = userMap.get(offenderAddress)!;
+		const timeDifference = currentTimeToSeconds() - user.lastTimeBridged;
+
+		//check timeDifference to make sure its not less than 16mins
+		// if it is then no modifications should be done as
+		// seperate events have the same information
+		if (timeDifference > hoursToSeconds(0.25)) {
```

Comment:
I think this should be using `BRIDGE_TIME_PERIOD` instead of 16 minutes.

Also we should be using a running window to expire older bridge transactions from the user.

Let's say that `BRIDGE_TIME_PERIOD == 24 hours` and `BRIDGE_LIMIT == 100_000 POLYX` and `timelock == 2 hours` with the following bridge transactions from the same user:

1. Day 1 at 6PM: 50_000 POLYX
2. Day 2 at 8AM: 10_000 POLYX -- Total bridged by user: 60_000 POLYX
3. Day 2 at 5PM: 50_000 POLYX -- Reached bridge limit in last 24 hour window.  The transaction will be rescheduled for retry later (based on `timelock`).
4. Day 2 at 7PM (2 hours later) retry transaction from step `3`, the transaction can now go through, because the first 50_000 POLYX from step 1 is expired (outside of the running `BRIDGE_TIME_PERIOD` window)

I recommend that `userMap` stores an array of `userInfo` for each user.  New bridge transaction for the user will get appended to that array.  Expired transactions (`user.lastTimeBridged > BRIDGE_TIME_PERIOD`) will be removed/filtered from that array.  Then the array would only have transactions valid for the current running window `BRIDGE_TIME_PERIOD`.

A sum of the `amount` from still valid transaction would be the balance to check against `BRIDGE_LIMIT`.